### ISSUE TITLE
Immutable pointer should use `const Type*` in C

### ISSUE THREAD

**Issue by blackshirt** [vlang/v#8084]
i have a problem with vlang version 0.2 when trying to run examples file from examples/database/sqlite.v
I have run successfully on version 0.1.3 before.
here the generated messages (i'm on win64 bit, with mingw64 installed)
```
`C:\Users\regular\Downloads\v_windows\v\examples\database>v -cg run sqlite.v
Compilation with tcc failed. Retrying with gcc ...
C:\Users\regular\AppData\Local\Temp\v\sqlite.4932403150175385109.tmp.c: In function 'sqlite__DB_q_string':
C:\Users\regular\AppData\Local\Temp\v\sqlite.4932403150175385109.tmp.c:8566:25: warning: passing argument 1 of 'tos_clone' discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
  string res = tos_clone(sqlite3_column_text(stmt, 0));
                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:\Users\regular\AppData\Local\Temp\v\sqlite.4932403150175385109.tmp.c:6917:26: note: expected 'byteptr' {aka 'unsigned char *'} but argument is of type 'const unsigned char *'
 string tos_clone(byteptr s) {
                  ~~~~~~~~^
C:\Users\regular\AppData\Local\Temp\v\sqlite.4932403150175385109.tmp.c: In function 'sqlite__DB_exec':
C:\Users\regular\AppData\Local\Temp\v\sqlite.4932403150175385109.tmp.c:8584:27: warning: passing argument 1 of 'tos_clone' discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
    string val = tos_clone(sqlite3_column_text(stmt, i));
                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:\Users\regular\AppData\Local\Temp\v\sqlite.4932403150175385109.tmp.c:6917:26: note: expected 'byteptr' {aka 'unsigned char *'} but argument is of type 'const unsigned char *'
 string tos_clone(byteptr s) {
                  ~~~~~~~~^
C:\Users\regular\.vmodules\cache\91\914f3bd92e13c1ad3ab0196f8011d0c9.o: In function `sqlite3_win32_write_debug':
<command line>:(.text+0xf9f9): undefined reference to `__chkstk'
C:/Program Files/mingw-w64/x86_64-8.1.0-posix-seh-rt_v6-rev0/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/8.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\Users\regular\Downloads\v_windows\v\examples\database\sqlite.exe(.pdata): relocation ".uw_base+0x0 (type R_X86_64_RELATIVE)" goes out of range
C:/Program Files/mingw-w64/x86_64-8.1.0-posix-seh-rt_v6-rev0/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/8.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: final link failed: Symbol needs debug section which does not exist
collect2.exe: error: ld returned 1 exit status
builder error:
==================
C error. This should never happen.

If you were not working with C interop, please raise an issue on GitHub:

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang`
```
any help very appreciated...thanks

**Comment by JalonSolov** [comment]
Adding 3 backticks before and after output makes it _much_ easier to read.  I edited your message to show this.

**Comment by blackshirt** [comment]
thanks for correction.
I think the problem happen only on windows, because in my linux box (ubuntu 20.04) the problem does not appears
here the output from the linux box
'''
green@home-laptop1:~$ v up
Updating V...
Current V version:
V 0.2.1 c8bcee9, timestamp: 2021-01-14 02:21:27 +0100
green@home-laptop1:~$ cd v
green@home-laptop1:~/v$ cd examples/
green@home-laptop1:~/v/examples$ cd database/
green@home-laptop1:~/v/examples/database$ ls
mysql.v  pg  sqlite.v
green@home-laptop1:~/v/examples/database$ v run sqlite.v
nr users = 3
SQL Result code: 101
['1', 'Sam']
['2', 'Peter']
['3', 'Kate']
green@home-laptop1:~/v/examples/database$
'''

Any clues for the problems ?

**Comment by ntrel** [comment]
cgen could prefix parameters with const when they're not mut in V. This should be done for variables and fields too. That could lead to better optimization as well.

**Comment by blackshirt** [comment]
> cgen could prefix parameters with const when they're not mut in V. This should be done for variables and fields too. That could lead to better optimization as well.

actually, i dont know bit detail how cgen works internally for now

**Comment by blackshirt** [comment]
**Updated**
 i think its happens only on my windows, see the output :
```
C:\Users\reguler>v version
V 0.2.1 1559e72

C:\Users\reguler>v
Welcome to the V REPL (for help with V itself, type `exit`, then run `v help`).
V 0.2.1 1559e72
Use Ctrl-C or `exit` to exit, or `help` to see other available commands
>>> import sqlite
==================
C:/Program Files/mingw-w64/x86_64-8.1.0-posix-seh-rt_v6-rev0/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/8.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: final link failed: Symbol needs debug section which does not exist
collect2.exe: error: ld returned 1 exit status
...
==================
(Use `v -cg` to print the entire error message)

builder error:
==================
C error. This should never happen.

If you were not working with C interop, please raise an issue on GitHub:

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
>>>
```

**Comment by spaceface777** [comment]
Fixed in vlang/tccbin@bac32fd.
